chapters/classes.tex

```diff
@@ -351,62 +373,98 @@ \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings
+
+\begin{example}
+\begin{lstlisting}[language=modelica]
+  parameter Integer level(min=1)=1;
+  Motor motor;
+  Level1 component1(J=J) if level==1 "Conditional component";
+  Level2 component2 if level==2 "Conditional component";
+  Level3 component3(J=component1.J) if level<2 "Conditional component";
+  // Illegal modifier on component3 since component1.J is conditional
+  // Even if we can see that component1 always exist if component3 exist
+equation
+  connect(component1$\ldots$, $\ldots$) "Connection to conditional component 1";
+  connect(component2.n, motor.n) "Connection to conditional component 2";
+  connect(component3.n, motor.n) "Connection to conditional component 3";
+  component1.u=0; // Illegal
+\end{lstlisting}
+\end{example}
+
+The \emph{expression} must be a \lstinline!Boolean! scalar expression, and must be a parameter expression.
+
+\begin{nonnormative}
+A parameter expression is required since it shall be evaluated at compile time.
+\end{nonnormative}
+
+A redeclaration of a component shall not include a condition attribute;
+and the condition attribute is kept from the original declaration (see
+\cref{interface-compatibility-or-subtyping}).
+
+If the \lstinline!Boolean! expression is false the component (including its modifier) is removed from the flattened DAE, and connections to/from the component are removed.
+A component declared with a condition-attribute can only be modified and/or used in connections.
+
+\begin{nonnormative}
+Adding the component and then removing it ensures that the component is valid.
+
+If a connect equation defines the connection of a non-conditional component \lstinline!c1! with a conditional component \lstinline!c2! and \lstinline!c2! is de-activated, then \lstinline!c1! must still be a declared element.
+
+There are annotations to handle the case where the connector should be connected when activated, see \cref{annotations-for-the-graphical-user-interface}.
+\end{nonnormative}
+
+
+\section{Component Variability}\label{component-variability}
+
+As briefly mentioned in \cref{component-variability-prefixes-discrete-parameter-constant}, the component variability prefixes are the basis for defining \firstuse{component variability}\index{declared variability}\index{variability!declared|see{declared variability}}.
+Combined with some other information about the components and analysis of expression variability (\cref{variability-of-expressions}), they define the component variabilities as follows:
 \begin{itemize}
 \item
-  A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix remains constant during simulation, with a value that is unaffected even by the initialization problem.
+  A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix does not change during simulation, with a value that is unaffected even by the initialization problem (i.e., determined during translation).
```

Comment:
I would drop "determined during translation":
```suggestion
  A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix does not change during simulation, with a value that is unaffected even by the initialization problem.
```
Since one could have it bound to an external function, that is evaluated as part of running the model (seen as before initialization at least in Dymola).

I realize that with the change of including constants as part of evaluable, it must be possible to evaluate it - but there's still no need to _actually evaluate it_ during translation.

Reply:
It says _determined_, not _evaluated_, so I think that the clarification is helpful without being misleading.

Reply:
I take the approving review as a sign of this being OK as is.  Resolving.